Marlin/Cartridge.cpp

```diff
@@ -76,24 +75,58 @@ static void updateCartridgeStatus(void);
  * @returns    Returns true if a cartridge has been removed, or no 
  *             cartridge is present WITH HYSTERESIS
  */
-bool CartridgeRemoved(void) {
-    bool returnValue = false;
-    static uint8_t cartridgeRemovalHysteresis = 0;
 
-    // Update Cartridge Status so that we have fresh information for the 
-    // function
-    updateCartridgeStatus();
+ /**
+ * Check to see if cartridges are present or absent. Flags internally if 
+ * one has been removed, or clears the removed flag if it's present. 
+ * The status of cartridge removal can be found with 
+ * CartridgeRemoved()
+ */
+void UpdateCartridgeStatus(void) {
+    if (cartridgeRemovalCheckEnabled == true) {
+        // Cartridge zero is pulled low by default
+        if (READ(CART0_SIG2_PIN) == HIGH) {
+            cartridgePresentUpdate(0);
+        } else {
+            cartridgeAbsentUpdate(0);
+        }
+        // Cartridge one is pulled high by default
+        if (READ(CART1_SIG2_PIN) == LOW) {
+            cartridgePresentUpdate(1);
+        } else {
+            cartridgeAbsentUpdate(1);
+        }
+    }
+    else {
+        cartridgePresentUpdate(0);
+        cartridgePresentUpdate(1);
+    }
+}
+
 
-    // Set up addition condition beyond cartridgeRemovedCheck that
-    // could result in an the cartridge needing to be reported as missing.
-    bool removedCondition =
-        (cartridgeStatus[FFF_INDEX] != PRESENT);
+/**
+ * This function checks to see if the FFF cartridge is removed,
+ * to prevent heating
+ * @returns    Returns true if an FFF cartridge has been removed
+ */
+bool CartridgeRemovedFFF(void) {
+    UpdateCartridgeStatus();
+    return (cartridgeStatus[FFF_INDEX] != PRESENT);
+}
```

Comment:
I think this function is not necessary and adds confusion + multiple function calls. 
Instead, I would replace this function with (cartridgeStatus[FFF_INDEX] != PRESENT)